compiler/luci/service/src/CircleShapeInferenceRule.cpp

```diff
@@ -607,6 +607,7 @@ loco::NodeShape infer_depthwise_conv2d(const luci::CircleDepthwiseConv2D *node)
   assert(ifm_shape.rank() == 4);
   assert(ker_shape.rank() == 4);
   assert(ker_shape.dim(0).value() == 1);
+  assert(ifm_shape.dim(3).value() * node->depthMultiplier() == ker_shape.dim(3).value());
```

Comment:
Not sure this is right place to be. Is there stage to shape check for each circle op?

Reply:
I think `bool CircleDepthwiseConv2DGraphBuilder::validate(const ValidateArgs &args) const` would be the place.

Reply:
Condition check here is also good.

Reply:
> Condition check here is also good.
:+1: 

To cover optimization process, both are needed.

Reply:
> I think `bool CircleDepthwiseConv2DGraphBuilder::validate(const ValidateArgs &args) const` would be the place.

Thanks, I would do when required :)